File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        Set<Integer> uniques = new HashSet<>();

        int size = uniques.size();
        System.out.println(size);

        for (int num : nums) {
            uniques.add(num);
            if (size == uniques.size()) {
                return true;
            }
            size = uniques.size();
        }

        return false;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        ans = []

        #prefix
        ans.append(1)
        pref = 1
        for i, num in enumerate(nums):
            if i == len(nums)-1:
                break
            pref *= num
            ans.append(pref)
    
        #postfix
        #[1,2,4,6]
        #[1,1,2,8]
        #[48,24,6,1]
        postfix = 1
        for i in range(len(nums) - 1, -1, -1):
            ans[i] *= postfix
            postfix *= nums[i]
        return ans
